File: 2705_compact_object/Solution.cs
// LeetCode 2705 - Compact Object
// https://leetcode.com/problems/compact-object/

// JS compactObject stand-in for int vectors: drop zeros
using System.Collections.Generic;

public class Solution {
    public int[] CompactObject(int[] obj) {
        var outList = new List<int>();
        foreach (int x in obj) if (x != 0) outList.Add(x);
        return outList.ToArray();
    }
}

File: 2705_compact_object/solution.js
// LeetCode 2705 - Compact Object
// https://leetcode.com/problems/compact-object/

var compactObject = function(obj) {
    if (Array.isArray(obj)) {
        const out = [];
        for (const x of obj) {
            const v = compactObject(x);
            if (v) out.push(v);
        }
        return out;
    }
    if (obj !== null && typeof obj === "object") {
        const out = {};
        for (const k of Object.keys(obj)) {
            const v = compactObject(obj[k]);
            if (v) out[k] = v;
        }
        return out;
    }
    return obj;
};
